x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/form_validation/validation.helpers.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TestBedConfig } from '@kbn/test/jest';
+import {
+  createColdPhaseActions,
+  createDeletePhaseActions,
+  createErrorsActions,
+  createFormSetValueAction,
+  createFormToggleAction,
+  createFrozenPhaseActions,
+  createHotPhaseActions,
+  createRolloverActions,
+  createSavePolicyAction,
+  createTogglePhaseAction,
+  createWarmPhaseActions,
+} from '../../helpers';
+import { initTestBed } from '../init_test_bed';
+
+type SetupReturn = ReturnType<typeof setupValidationTestBed>;
+
+export type ValidationTestBed = SetupReturn extends Promise<infer U> ? U : SetupReturn;
+
+export const setupValidationTestBed = async (arg?: { testBedConfig?: Partial<TestBedConfig> }) => {
+  const testBed = await initTestBed(arg);
+
+  return {
+    ...testBed,
+    actions: {
+      togglePhase: createTogglePhaseAction(testBed),
+      setPolicyName: createFormSetValueAction(testBed, 'policyNameField'),
+      savePolicy: createSavePolicyAction(testBed),
+      toggleSaveAsNewPolicy: createFormToggleAction(testBed, 'saveAsNewSwitch'),
+      ...createRolloverActions(testBed),
+      ...createErrorsActions(testBed),
+      ...createHotPhaseActions(testBed),
+      ...createWarmPhaseActions(testBed),
+      ...createColdPhaseActions(testBed),
+      ...createFrozenPhaseActions(testBed),
+      ...createDeletePhaseActions(testBed),
+    },
```

Comment:
I think a better solve for this might have been a factory function that would accept an array of action creators, but such a function would not give me the correct inferred types. I'm curious if anyone has a good idea for this.